packages/block-library/src/navigation/index.php

```diff
@@ -384,20 +384,31 @@ function block_core_navigation_get_most_recently_published_navigation() {
 }
 
 /**
- * Filter out empty "null" blocks from the block list.
- * 'parse_blocks' includes a null block with '\n\n' as the content when
+ * Recursively filter out blocks from the block list that are considered invalid inside Navigation.
+ * This list includes:
+ * - The Navigation block itself (results in recursion).
+ * - empty "null" blocks from the block list.
+ *
+ * Note: 'parse_blocks' includes a null block with '\n\n' as the content when
  * it encounters whitespace. This is not a bug but rather how the parser
  * is designed.
  *
- * @param array $parsed_blocks the parsed blocks to be normalized.
- * @return array the normalized parsed blocks.
+ * @param array $parsed_blocks the parsed blocks to be filtered.
+ * @return array the filtered parsed blocks.
  */
-function block_core_navigation_filter_out_empty_blocks( $parsed_blocks ) {
-	$filtered = array_filter(
+function block_core_navigation_filter_out_invalid_blocks( $parsed_blocks ) {
+	$filtered = array_reduce(
 		$parsed_blocks,
-		function( $block ) {
-			return isset( $block['blockName'] );
-		}
+		function( $carry, $block ) {
+			if ( isset( $block['blockName'] ) && 'core/navigation' !== $block['blockName'] ) {
```

Comment:
Should we use an array of allowedBlocks here? There's [this array](https://github.com/WordPress/gutenberg/blob/785db7dc03d8a0d8e256a7c170d4cf384ed17685/packages/block-library/src/navigation/edit/inner-blocks.js#L18).

Reply:
Updated...but is there any danger of breaking sites that might utilise 3rd party blocks here?